### PR TITLE
Move codeblock to sphinx php repository

### PR DESCRIPTION
**Pull request by WouterJ**

### CODE REVIEW

**Comment by xabbuh** [comment]
Isn't it possible to reuse some code of the `CodeBlock` class from the Sphinx core?

File: sensio/sphinx/codeblock.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
+from docutils.parsers.rst import Directive, directives
+from docutils import nodes
+from sphinx.util.nodes import set_source_info
+
+"""
+The only change with the built-in CodeBlock class that comes with Sphinx
+is that linenos is always on.
```
Comment (stof) [inline sensio/sphinx/codeblock.py]: that's not true. this is the only change compared to Sphinx 1.2. But compared to Sphinx 1.3, it also removes all new features

**Comment by WouterJ** [comment]
I think I have an idea how to make this much better, will test & commit when I'm home

**Comment by WouterJ** [comment]
Updated with a much better implementation, it works locally.

**Comment by xabbuh** [comment]
:+1: looks good

File: sensio/sphinx/codeblock.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
+from sphinx.directives.code import CodeBlock
+
+"""
+The only change with the built-in CodeBlock class that comes with Sphinx
+is that linenos is always on.
+"""
+class NumberedCodeBlock(CodeBlock):
+    def run(self):
+        literal = super(NumberedCodeBlock, self).run()[0];
+        literal['linenos'] = 'linenos'
```
Comment (stof) [inline sensio/sphinx/codeblock.py]: shouldn't it be a boolean value ?

Comment (WouterJ) [inline sensio/sphinx/codeblock.py]: No, but linenos is not correct as well:

 > `linenos`
 >If set to ``'table'``, output line numbers as a table with two cells,
 >one containing the line numbers, the other the whole code.  This is
 >copy-and-paste-friendly, but may cause alignment problems with some
 >browsers or fonts.  If set to ``'inline'``, the line numbers will be
 >integrated in the ``<pre>`` tag that contains the code (that setting
 >is *new in Pygments 0.8*).

 >For compatibility with Pygments 0.7 and earlier, every true value
 >except ``'inline'`` means the same as ``'table'`` (in particular, that
 >means also ``True``).

 >The default value is ``False``, which means no line numbers at all.

I'll change it to table, as that seems what is used by symfony.com.

**Comment by WouterJ** [comment]
I've now made the code even more simple (and changed the value). Didn't test it yet, will test it this afternoon.

File: sensio/sphinx/codeblock.py (hunk 2 above)
Comment (stof) [inline sensio/sphinx/codeblock.py]: The codebase of Sphinx seems to set the value to ``True``.

**Comment by stof** [comment]
:+1:

File: sensio/sphinx/codeblock.py (hunk 2 above)
Comment (WouterJ) [inline sensio/sphinx/codeblock.py]: Yeah, true is supported by Pygments 0.7 - 2.0 (for BC reasons). Not sure if we want that.

**Comment by WouterJ** [comment]
Tested and seem to work perfect. Can be merged @fabpot

**Comment by fabpot** [comment]
Thank you @WouterJ.

**Pull request closed without merging**